extension-readme.md

```diff
@@ -82,53 +87,66 @@ For complete setup instructions for these three components, see [Installing the
--   List and delete AWS CloudFormation stacks
--   List, invoke, and delete [AWS Lambda](https://aws.amazon.com/lambda/) functions
+After you've [created a serverless application](https://docs.aws.amazon.com/console/toolkit-for-vscode/create-sam), you can locally run and debug it using a SAM template. This YAML-formatted file describes resources such as functions, APIs, databases, and event-source mappings that make up a serverless application. 
 
-The following example shows a highlight of this functionality.
+To start debugging with a SAM template, choose `Add Debug Configuration` from the _CodeLens_ indicator in the template file.    
 
-After you have [created a serverless application](https://docs.aws.amazon.com/console/toolkit-for-vscode/create-sam), you can locally run, debug, and further develop that application in the VS Code editor. For example:
+![Add Debug Configuration Template](./resources/marketplace/Codelens-YAML-template.png)
+###### The _CodeLens_ indicator in the SAM template allows you to add a debug configuration for the serverless application.</h6>
 
-![Configure and Run 1](./resources/marketplace/sam-configure-and-run-still-1.png)
+Alternatively, you can run and debug just the AWS Lambda function and exclude other resources defined by the SAM template. Again, use a _CodeLens_ indicator for an AWS Lambda-function handler. (A _handler_ is a function that Lambda calls to start execution of a Lambda function.) 
 
-Take special note of the _CodeLenses_ that enable you to run and debug the application locally.
+![Add Debug Configuration Direct](./resources/marketplace/Codelens-direct-function.png)
 
-The status and results are shown in the **OUTPUT** panel when the **AWS Toolkit** output channel is selected.
+###### The _CodeLens_ indicator in the application file lets you add a debug configuration for a selected AWS Lambda function.
+
+
+When you run a debug session, the status and results are displayed in the **OUTPUT** panel when the **AWS Toolkit** output channel is selected.
 
 ![Configure and Run 1](./resources/marketplace/sam-configure-and-run-still-2.png)
 
-When you're satisfied, you can [deploy your application](https://docs.aws.amazon.com/console/toolkit-for-vscode/deploy-serverless-app) to a CloudFormation stack and then [run the Lambda function](https://docs.aws.amazon.com/console/toolkit-for-vscode/remote-lambda) on AWS.
+###### After a local run is complete, the output appears in the **OUTPUT** tab.
+
+
+When you're satisfied with performance, you can [deploy your serverless application](https://docs.aws.amazon.com/console/toolkit-for-vscode/deploy-serverless-app). During deployment, the SAM template is converted to a CloudFormation template, which is then used to configure all your application's resources in the AWS Cloud. 
+
+### Additional information about CodeLenses 
+
+The AWS Toolkit for Visual Studio Code supports _CodeLens_-enabled debugging for the following languages and runtimes: 
```

Comment:
You can probably remove `_CodeLens_-enabled` or reword, you can debug these languages regardless of whether or not codelenses are drawing.